src/shared/components/plots/MultipleCategoryHeatmap.tsx

```diff
@@ -141,33 +140,38 @@ function toHeatmapTracks(
         []
     );
 
-    const totalSamplesPerGroup = result.map(t =>
-        t.data
-            .map(d => d.profile_data)
-            .reduce((partialSum: number, d: number) => partialSum + d, 0)
-    ) as number[];
+    const groupsPercentages: number[][] = [];
+
+    groups.forEach(group => {
+        const groupSampleCount =
+            group.data
+                .map(d => d.profile_data)
+                .reduce((partialSum: number, d: number) => partialSum + d, 0) ||
+            0;
+        const groupPercentages: number[] = [];
+        groupsPercentages.push(groupPercentages);
+        group.data.forEach(d => {
+            const sampleCount = d.profile_data || 0;
+            const percentage = (sampleCount / groupSampleCount) * 100;
+            groupPercentages.push(percentage);
+            // Tooltip label:
+            d.sample = `${
+                d.sample
+            }; ${sampleCount} samples (${percentage.toFixed(1)}%)`;
+        });
+    });
 
-    const maxSamples = Math.max(
-        ...([] as any).concat(
-            ...result.map(t => t.data.map(d => d.profile_data))
-        )
+    // Calculate 'heat' of heatmap cell:
+    const percentageMax = Math.max(
+        ...groupsPercentages.reduce((group, result) => result.concat(group), [])
```

Comment:
i think flatten would work here?   flatten is native array function and also there's lodash flatten if necessary

Reply:
I had tried flatmap, but both flatMap and flat are not available with the current typescript compiler settings. But I'll try lodash instead

Reply:
Fixed in [4e108f5](https://github.com/cBioPortal/cbioportal-frontend/compare/8800d778f277a681acfaaeccc20c4afb87206467..4e108f582b55e15de815e2fec03d53b8905004e7)